aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -259,13 +261,23 @@ private boolean isGlueIcebergTable(Table table) {
   public boolean dropTable(TableIdentifier identifier, boolean purge) {
     try {
       TableOperations ops = newTableOps(identifier);
-      TableMetadata lastMetadata = ops.current();
+      TableMetadata lastMetadata;
+      try {
+        lastMetadata = ops.current();
```

Comment:
this will keep retrying because it will only fail with `NoSuchTableException`, so `ops.current()` should not be called.